kong/runloop/balancer.lua

```diff
@@ -835,127 +911,49 @@ end
 
 
 local function init()
-  if kong.configuration.worker_consistency == "eventual" then
-    local opts = { neg_ttl = 10 }
-    local upstreams_dict, err = singletons.core_cache:get("balancer:upstreams",
-                                        opts, load_upstreams_dict_into_memory)
-    if err then
-      log(CRIT, "failed loading list of upstreams: ", err)
-      return
-    end
-
-    for _, id in pairs(upstreams_dict) do
-      local upstream_cache_key = "balancer:upstreams:" .. id
-      local upstream, err = singletons.core_cache:get(upstream_cache_key, opts,
-                      load_upstream_into_memory, id)
-
-      if upstream == nil or err then
-        log(WARN, "failed loading upstream ", id, ": ", err)
-      end
-
-      local target_cache_key = "balancer:targets:" .. id
-      local target, err = singletons.core_cache:get(target_cache_key, opts,
-                load_targets_into_memory, id)
-      if target == nil or err then
-        log(WARN, "failed loading targets for upstream ", id, ": ", err)
-      end
-    end
+  if kong.configuration.worker_consistency == "strict" then
+    create_balancers()
+    return
```

Comment:
I know this is not about this PR, but I have one question about this.

With `eventual` when it was first introduced for `router` and `plugin iterator` rebuilds, it meant `only eventual`. That also meant that it was `always on`. Things were prepared in a background. `strict` meant that on request time we do check if those are up to date, that check we don't do with `eventual`, thus the `only eventual`.

I am not sure does the balancer take same approach, but I think it should if it is possible.